### PR TITLE
Fix for issue 14265

### PR DESCRIPTION
**Pull request by ruudboon**
Fixed issue #14265, added test and added method to get Path.

### CODE REVIEW

File: CHANGELOG-4.0.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # [4.0.0-beta.2](https://github.com/phalcon/cphalcon/releases/tag/v4.0.0-beta.2) (2019-xx-xx)
 
+# Added
+- Added `Phalcon\Session\Adapter\Stream::getPath()` method that returns current save path.
```
Comment (sergeyklay) [inline CHANGELOG-4.0.md]: The PR comment says that proposed changes fixes the bug described in #14265. Should we mention this as well?

Comment (ruudboon) [reply CHANGELOG-4.0.md]: Absolutely. Will add this

**Comment by sergeyklay** [comment]
@ruudboon I would like suggest something like:
```diff
diff --git a/phalcon/Session/Adapter/Stream.zep b/phalcon/Session/Adapter/Stream.zep
index cb6d649f7..78da958a0 100644
--- a/phalcon/Session/Adapter/Stream.zep
+++ b/phalcon/Session/Adapter/Stream.zep
@@ -54,10 +54,13 @@ class Stream extends Noop
          */
         if !fetch path, options["savePath"] {
             let path = ini_get("session.save_path");
+            if empty path {
+                let path = sys_get_temp_dir();
+            }
         }
 
         if unlikely !is_writable(path) {
-            throw new Exception("The save_path [" . path . "]is not writeable");
+            throw new Exception("The save_path [" . path . "] is not writeable");
         }
 
         let this->path = Str::dirSeparator(path);
@@ -69,7 +72,9 @@ class Stream extends Noop
 
         let file = this->path . this->getPrefixedName(id);
 
-        if file_exists(file) && is_file(file) {
+        // is_file() will return FALSE if the given path points to
+        // a directory or there is no file.
+        if is_file(file) {
             unlink(file);
         }
 
@@ -84,9 +89,9 @@ class Stream extends Noop
             time    = time() - maxlifetime;
 
         for file in glob(pattern) {
-            if file_exists(file) &&
-               is_file(file)     &&
-               (filemtime(file) < time) {
+            // is_file() will return FALSE if the given path points to
+            // a directory or there is no file.
+            if is_file(file) && (filemtime(file) < time) {
                 unlink(file);
             }
         }
@@ -96,14 +101,14 @@ class Stream extends Noop
 
     public function open(var savePath, var sessionName) -> bool
     {
-        var path;
-
-        if true !== ends_with(savePath, "/") {
-            let path = savePath . "/";
+        if unlikely !empty savePath {
+            if true !== ends_with(savePath, "/") {
+                let this->path = savePath . "/";
+            } else {
+                let this->path = savePath;
+            }
         }
 
-        let this->path = path;
-
         return true;
     }
```

Also I'm not sure about getPath. Do you need it only for testing purposes? Is it possible to achieve the same result using reflection? 
https://github.com/phalcon/cphalcon/blob/a0b25edcfc143b780ec69d5b92cae9c0c37c1987/tests/_support/Helper/Unit.php#L103-L112

**Comment by CameronHall** [comment]
@ruudboon can you please try rebasing instead of merging?

**Pull request closed without merging**